projects/client/RabbitMQ.Client/src/client/impl/AutorecoveringConnection.cs

```diff
@@ -1069,9 +1084,11 @@ private async Task RecoveryLoopConnectedHandler(RecoveryCommand command)
                     break;
                 case RecoveryCommand.BeginAutomaticRecovery:
                     _recoveryLoopState = RecoveryConnectionState.Recovering;
-                    await Task.Delay(_factory.NetworkRecoveryInterval).ConfigureAwait(false);
-                    _recoveryLoopCommandQueue.Enqueue(RecoveryCommand.PerformAutomaticRecovery);
-                    _semaphore.Release();
+                    Task.Delay(_factory.NetworkRecoveryInterval)
```

Comment:
Without an await or a task being assigned to the result, this is not guaranteed to run.

Reply:
I always try to see things in the TPL land as "It is very likely I haven't fully understood things" that's why I considered your input to be true and wanted to hear more. No worries

Reply:
Me too, it's a massive rabbit hole, and assumptions (like mine) are almost always guaranteed to either be false or have a lot of "yeah but" cases.